### DISCUSSION TITLE
React, or Vue?

### DISCUSSION THREAD

**Discussion by gaocegege** [tensorchord/envd#562] [General]
/cc @sepush

We are investigating if we should migrate https://docusaurus.io/ to https://vitepress.vuejs.org/

I think the question we should ask is if we want react, or vue.

We will make an MVP for vitepress to see if it works for us.

**Comment by gaocegege** [comment]
The benefit of vue.js from my side:

- Community: There may be more vue.js engineers here
- Speed: ![](https://cdn.discordapp.com/attachments/985713239738449951/993471409953185792/vitepress.gif)

**Reply by VoVAllen** [reply]
I believe live editing is more likely a site generator function instead of builtin functionality of react/vue.  Docusaurus seems to support this in v1 https://github.com/facebook/docusaurus/pull/663.

**Reply by gaocegege** [reply]
docusaurus live updating is a little slower than vitepress, I think.

**Comment by hwdef** [comment]
react +1

**Reply by gaocegege** [reply]
Could you please explain more about it? Why do you vote for react :thinking:

**Reply by hwdef** [reply]
It's just some personal reasons.
They are all perfect,  no matter which one you choose, it's actually the same.

**Comment by Sepush** [comment]
### VitePress is build on top of Vite

- Faster dev server start
- Faster hot updates
- Faster build (uses Rollup internally)

### Lighter page weight

Vue 3 tree-shaking + Rollup code splitting

Does not ship metadata for every page on every request. This decouples page weight from total number of pages. Only the current page's metadata is sent. Client side navigation fetches the new page's component and metadata together.


### Others
It is suitable for out-of-the-box documentation use, but the config and theming API may still change between minor releases.

Easy to custom the theme through It's config and theme API.

We could easily build a beautiful doc site via VitePress.
It will looks like:
https://main.vitejs.dev/
https://vitest.dev/

**Reply by Sepush** [reply]
I will give a demo later this evening.

**Reply by Sepush** [reply]
Still WIP
https://github.com/Sepush/envd-docs/tree/vitepress
You can git clone this repo and run the following commond to preview

```sh
// If you did not use pnpm.It worth a try.

// install pnpm
npm i -g pnpm
// install dependencies
pnpm i
// run to dev
pnpm docs:dev

```

**Reply by Sepush** [reply]
It's better than the old version
![image](https://user-images.githubusercontent.com/39197136/177531736-cb30270a-2119-45a3-ba62-478c07d3c2da.png)
![image](https://user-images.githubusercontent.com/39197136/177531857-a1e3a572-147f-40a6-8419-7fd8582b36b1.png)
![image](https://user-images.githubusercontent.com/39197136/177532016-c73575e7-35a1-445f-95b4-d742203b4297.png)

**Reply by gaocegege** [reply]
Looking forward to it!

**Comment by STRRL** [comment]
tldr; I would prefer vitepress rather than docusaurus.

> We are investigating if we should migrate [docusaurus.io](https://docusaurus.io/) to [vitepress.vuejs.org](https://vitepress.vuejs.org/)
I think the question we should ask is if we want react, or vue.

I do not think the origin question(docusaurus or vitepress) would leads to the second question(react or vue). Because it seems we are going to build the homepage of envd, it looks like more about a "static site" than an "interactive single-page-application". For building a single page application, I prefer to use React, but for building static site, I prefer to use vitepress at the next time.

I did not use vitepress before, but I prefer to give it a try. We(chao mesh team) are using docusaurus as the tool for generating documentation site, I think docusaurus might be a little hard to configure with both multi-versoning mechanism and i18n. And vitepress and vitepress community have several solution for that: 

- https://github.com/vuejs/vuepress/issues/1018#issuecomment-440248700
- https://vuepress.vuejs.org/guide/i18n.html#site-level-i18n-config

**Reply by gaocegege** [reply]
Thanks for your reply!

**Comment by VoVAllen** [comment]
I investigated a little bit about vue vs react. For the website adoption, the ratio of Vue vs React is 1:1~1:2. Therefore there's no significant gap between them. 
For the UI systems, major UI frameworks support both. There're some frameworks only specific framework, but no significant difference in terms of quality or quantity. 
For the whole ecosystem, React is bit stronger than Vue, since backed by several company (Meta, vercel, etc.). While vue ecosystem is by a group of individual contributor, for example React SWR vs VueRequest.

Vue is easier for newbies and seems simpler. I personally favor Vue. And @Sepush 's demo looks really nice. One problem is that vitepress seems still in an early stage of development. Is it stable enough now?

**Comment by shaonianche** [comment]
I believe docusaurus.io or vitepress.vuejs.org can support our work very well.
But for now we have an online version of react, and it doesn't have problems that can't be solved.
I think we should spend some more time on the documentation content ...

**Reply by gaocegege** [reply]
Yep, I totally agree. The content is more important than the framework. 

While Sepush@ can help us migrate to vitepress, it does not break the progress to improve the content, I think.

**Reply by shaonianche** [reply]
> 
If no one on the current team is good at front-end, then I think it's good to bring in developers and frameworks that are good at it.

**Comment by gaocegege** [comment]
https://62c7a984e3f8da00081f5046--inspiring-kulfi-9ab7b2.netlify.app/ The vitepress preview looks promising!

Thank @Sepush for the work!

**Comment by gaocegege** [comment]
Finally, we moved from docusaurus to vitepress. https://envd.tensorchord.ai/

See https://twitter.com/TensorChord/status/1554286590130028544

Thank @Sepush
